### DISCUSSION TITLE
Message Bulk Deleting

### DISCUSSION THREAD

**Discussion by TryhardOne** [Discord4J/Discord4J#903] [Q&A] [answered]
Hello,
I was trying to implement a way of deleting multiple messages from a channel.
Unfortunately, by doing this the normal way with using `Message.delete()` this action is limited by the Discord APIs rate limiter.

The only thing I found related to this was MessageDeleteBulk
I tried using 
```
MessageDeleteBulk.builder().addAllIds(test).channelId(msg.getChannelId().asString())
						.guildId(serverID).build();
```
where test is an arraylist containing the message ids.

However, this does not result in anything. I have no idea how to use this otherwise.
Thanks

**Comment by darichey** [comment, marked as answer]
Hi, you can use Discord's [bulk delete](https://discord.com/developers/docs/resources/channel#bulk-delete-messages) functionality with [`GuildMessageChannel::bulkDelete`](https://github.com/Discord4J/Discord4J/blob/4da58b43022ce455607e87de88a2e1fbe7246fe3/core/src/main/java/discord4j/core/object/entity/channel/GuildMessageChannel.java#L41-L85).

Take note of the warning about which messages qualify for bulk deleting.

**Reply by TryhardOne** [reply]
I read something about it, however, how would I get the GuildMessageChannel Object of a channel, or from a server?

**Reply by darichey** [reply]
You can use `Guild::getChannelById` or `GatewayDiscordClient::getChannelById` and then cast with `ofType`/`cast`. For example,
```java
gateway.getChannelById(Snowflake.of(myChannelId))
    .ofType(GuildMessageChannel.class)
```

**Reply by TryhardOne** [reply]
Ah nevermind, I can cast the MessageChannel to a GuildMessageChannel and use it like that, thanks :3
